dot-net-sdk/dto/bandit/ContextAttributes.cs

```diff
@@ -25,16 +25,46 @@ public ContextAttributes(string key)
         this.Key = key;
     }
 
-    public ContextAttributes(string key, Dictionary<string, object> other) {
+    public static ContextAttributes FromDict(string key, IDictionary<string, object?> other)
+    {
+        var obj = new ContextAttributes(key);
+        obj.AddDict(other);
+        return obj;
+    }
+
+    public static ContextAttributes FromNullableAttributes(string key, IDictionary<string, string?>? categoricalAttributes, IDictionary<string, double?>? numericAttributes)
```

Comment:
We allow the dev to pass dictionaries with possibly null values. .NET doesn't handle nullable reference types when they're used as a type parameter. i.e. `IDictionary<string, string>` cannot be cast to `IDictionary<string, string?>` even though it would be safe to do so.

Reply:
Ah interesting .NET struggles with it. We certainly want the users to pass null values (which we just ignore)